### PR TITLE
Fixed CaseInsensitiveXpath annotation

### PR DESCRIPTION
**Pull request by akamarouski**
The CaseInsensitiveXPath annotation allows you to convert specific parts of the xpath locator to case-insensitive.
Applies to the following types of locators:
 id
 name
 xpath
linktext
For all types of locators, except for xpath, before manipulation, there is a conversion to xpath.
The logic of work for WEB and MOBILE is different (the presence or absence of a link to the mobile application in Carina's configuration file depends on what logic is performed).
In the annotation, we can choose which parts of the locator will be transformed (id , name, text, class).
For WEB:
 id - @id attribute
 name - @name attribute
text - text() attribute
 class - @class attribute
For MOBILE:
id - @resource-id attribute
 name - @name attribute
text - attributes text(), @text, @content-desc
class - @class attributes

By default, all CaseInsensitiveXPath properties are disabled except for text.
PS When we convert linkText, there are text() attribute appear, so if you want to convert linkText, you should use text parameter of CaseInsensitiveXPath

### CODE REVIEW

File: carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/ExtendedElementLocator.java [hunk 1]
```diff
@@ -80,32 +78,27 @@ public ExtendedElementLocator(SearchContext searchContext, Field field) {
...
-            elements = searchContext.findElements(by);
-            if (elements.size() == 1) {
-                element = elements.get(0);
-            } else if (elements.size() > 1) {
-                element = elements.get(0);
-                LOGGER.debug(elements.size() + " elements detected by: " + by.toString());
-            }
-            
+
+        if (by == null) {
```
Comment (akamarouski) [inline carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/ExtendedElementLocator.java:82]: Change. nut by by places (null == by)

File: carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/ExtendedElementLocator.java [hunk 2]
```diff
@@ -80,32 +78,27 @@ public ExtendedElementLocator(SearchContext searchContext, Field field) {
...
+            throw new RuntimeException("By cannot be null");        // todo change to informative exception
```
Comment (akamarouski) [inline carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/ExtendedElementLocator.java:83]: NullPointerException(By cannot be null)

File: carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/LocatorConverter.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+    private static By idToXpathCaseInsensitive(String locator) {
+        String xpath = StringUtils.remove(locator, "By.id: ");
+        String attributePattern = "^.*$";
+
+        Matcher matcher = Pattern.compile(attributePattern).matcher(xpath);
+        StringBuilder sb = new StringBuilder();
+
+        while (matcher.find()) {
+            String attribute = "@id";
```
Comment (akamarouski) [inline carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/LocatorConverter.java:92]: check is mobile or not

Comment (akamarouski) [reply carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/LocatorConverter.java]: Done

File: carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/ExtendedElementLocator.java [hunk 4]
```diff
@@ -80,32 +89,27 @@ public ExtendedElementLocator(SearchContext searchContext, Field field) {
...
```
Comment (maksimsarychau) [inline carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/ExtendedElementLocator.java:94]: Throw NullPointer exception

Comment (akamarouski) [reply carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/locator/ExtendedElementLocator.java]: Done

**Pull request merged**